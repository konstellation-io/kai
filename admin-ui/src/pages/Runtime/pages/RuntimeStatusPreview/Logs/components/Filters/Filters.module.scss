@import '../../../../../../../styles/colors';
@import '../../../../../../../styles/mixins';

.container {
  @include font-body;
  display: flex;
  flex-direction: row;
  align-items: center;
  height: 48px;
}

.title {
  margin: 0 20px;
}

.filter {
  @include font-code;
  display: flex;
  flex-direction: row;
  padding: 3px 16px;

  background-color: $font-color-dark;
  color: $font-color-white;

  div:first-child {
    margin-right: 8px;
  }
}
