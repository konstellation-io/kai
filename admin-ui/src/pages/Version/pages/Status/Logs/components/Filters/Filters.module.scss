@import '../../../../../../../styles/colors';
@import '../../../../../../../styles/mixins';
@import '../../../../../../../styles/variables';

.container {
  @include border-header;
  @include font-body;
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: $grid-unit * 6;
  flex-shrink: 0;
}

.filter {
  @include font-code;
  display: flex;
  flex-direction: row;
  padding: 0 16px;

  background-color: $font-color-dark;
  color: $font-color-white;

  div:first-child {
    margin-right: 8px;
  }
}

.leftFilters {
  display: flex;
  align-items: center;
  margin-left: $grid-unit * 2.5;
}
.rightFilters {
  margin-right: $grid-unit * 2.5;
  display: flex;
  & > div:not(:last-child) {
    border-right: 2px solid #1d1e23;
    border-right-style: inset;
  }
}
.dateFilter {
  width: $grid-unit * 21.5;
}
