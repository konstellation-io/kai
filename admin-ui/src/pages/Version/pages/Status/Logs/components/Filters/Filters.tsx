import React from 'react';
import styles from './Filters.module.scss';
import DateFilter from './components/DatesFilter/DateFilter';
import { TabFilters } from '../../../../../../../graphql/client/queries/getLogs.graphql';
import Button from '../../../../../../../components/Button/Button';
import Right from '../../../../../../../components/Layout/Right/Right';

type Props = {
  onDateChange: Function;
  filterValues: TabFilters;
};
function Filters({ onDateChange, filterValues }: Props) {
  return (
    <div className={styles.container}>
      <Right style={styles.rightFilters}>
        <DateFilter
          selectedOption={filterValues.dateOption}
          onDateChange={onDateChange}
        />
        <Button label="JUMP TO NOW" />
      </Right>
    </div>
  );
}

export default Filters;
