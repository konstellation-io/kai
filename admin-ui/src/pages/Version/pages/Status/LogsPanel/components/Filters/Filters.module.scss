@import '../../../../../../../styles/colors';
@import '../../../../../../../styles/mixins';
@import '../../../../../../../styles/variables';

.container {
  @include font-body;
  line-height: 0;

  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: space-between;
  height: auto;
  padding: 0 20px;
  flex-shrink: 0;
}

.filterContainer {
  display: flex;
  flex-direction: row;
  align-items: center;

  > div {
    margin-right: $grid-unit;
  }
}
.rightFilters {
  flex-shrink: 0;

  .filterLevel {
    width: 142px;
  }

  .dateFilter {
    width: $grid-unit * 21.5;
  }
}

.leftContainer {
  flex-grow: 1;
  flex-basis: 0;
  max-width: 100 * $grid-unit;

  .searchFilter {
    width: 60%;
    height: 30px;
    min-width: 100px;
    margin: $grid-unit $grid-unit $grid-unit 0;

    .searchForm {
      input {
        height: 30px;
        min-height: 30px;
      }
    }
  }

  .selectProcesses {
    width: 40%;
    min-width: 170px;
  }
}
