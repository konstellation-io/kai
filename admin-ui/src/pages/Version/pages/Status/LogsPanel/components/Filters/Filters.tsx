import React from 'react';
import styles from './Filters.module.scss';
import DateFilter from './components/DatesFilter/DateFilter';
import Right from '../../../../../../../components/Layout/Right/Right';
import Left from '../../../../../../../components/Layout/Left/Left';
import SearchSelect from '../../../../../../../components/Form/SearchSelect/SearchSelect';
import MultiSelect, {
  MultiSelectOption,
  SelectTheme
} from '../../../../../../../components/Form/MultiSelect/MultiSelect';
import GroupSelect, {
  GroupSelectData
} from '../../../../../../../components/Form/GroupSelect/GroupSelect';
import cx from 'classnames';
import { LogLevel } from '../../../../../../../graphql/types/globalTypes';
import useWorkflowsAndNodes from '../../../../../../../hooks/useWorkflowsAndNodes';
import { NodeSelection } from '../../../../../../../graphql/client/typeDefs';
import { GetLogTabs_logTabs_filters } from '../../../../../../../graphql/client/queries/getLogs.graphql';
import LevelIcon from '../../../../../../../components/LevelIcon/LevelIcon';

// TODO: use GroupSelectData int he filters instead of [string]
function nodesSelectionToDoubleSelector(
  selections: NodeSelection[]
): GroupSelectData {
  const data: { [key: string]: string[] } = {};

  selections.forEach(({ workflowName, nodeNames }) => {
    data[workflowName] = nodeNames;
  });

  return data;
}

function doubleSelectorToNodesSelection(
  selections: GroupSelectData
): NodeSelection[] {
  const data: NodeSelection[] = Object.entries(selections).map(
    ([workflowName, nodeNames]) => ({
      workflowName,
      nodeNames,
      __typename: 'NodeSelection'
    })
  );

  return data;
}

const levelsOrdered = [
  LogLevel.INFO,
  LogLevel.DEBUG,
  LogLevel.WARN,
  LogLevel.ERROR
];

type Props = {
  versionId: string;
  updateFilters: Function;
  filterValues: GetLogTabs_logTabs_filters;
};
function Filters({ updateFilters, filterValues, versionId }: Props) {
  const { workflowsAndNodesNames } = useWorkflowsAndNodes(versionId);

  const levelOptions = levelsOrdered.map(
    (level: LogLevel) =>
      ({
        label: level,
        Icon: <LevelIcon level={level} />
      } as MultiSelectOption<LogLevel>)
  );

  function onNodeSelection(newSelection: GroupSelectData) {
    updateFilters({
      nodes: doubleSelectorToNodesSelection(newSelection)
    });
  }

  function onLevelSelection(newLevels: LogLevel[]) {
    updateFilters({ levels: newLevels });
  }

  function onSearchUpdate(newSearch: string) {
    updateFilters({ search: newSearch });
  }

  return (
    <div className={styles.container}>
      <Left className={cx(styles.filterContainer, styles.leftContainer)}>
        <div className={styles.searchFilter}>
          <SearchSelect
            options={[]}
            onChange={onSearchUpdate}
            value={filterValues.search}
            className={styles.searchForm}
            placeholder="Search"
            hideError
            hideLabel
            showSearchIcon
          />
        </div>
        <div className={styles.selectProcesses}>
          <GroupSelect
            options={workflowsAndNodesNames}
            formSelectedOptions={nodesSelectionToDoubleSelector(
              filterValues.nodes || []
            )}
            onChange={onNodeSelection}
            label=""
            placeholder="Select Processes"
            hideError
            hideSelections
          />
        </div>
      </Left>
      <Right className={cx(styles.filterContainer, styles.rightFilters)}>
        <div className={styles.filterLevel}>
          <MultiSelect<LogLevel>
            onChange={onLevelSelection}
            label=""
            hideError
            placeholder="ALL LEVELS"
            selectionUnit="LEVEL"
            selectAllText="ALL LEVELS"
            options={levelOptions}
            formSelectedOptions={filterValues.levels || []}
            className={styles.selectTypeForm}
            theme={SelectTheme.LIGHT}
          />
        </div>
        <DateFilter
          selectedOption={filterValues.dateOption}
          updateFilters={updateFilters}
          formStartDate={filterValues.startDate}
          formEndDate={filterValues.endDate}
        />
      </Right>
    </div>
  );
}

export default Filters;
