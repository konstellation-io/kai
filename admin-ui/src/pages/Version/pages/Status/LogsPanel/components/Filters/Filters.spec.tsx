import React from 'react';
import Filters, { Filter } from './Filters';
import { shallow } from 'enzyme';

describe('Filters', () => {
  let wrapper;

  const onDateChangeMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Filters
        onDateChange={onDateChangeMock}
        filterValues={{
          startDate: '2020-01-01',
          endDate: '2020-01-01',
          dateOption: '2020-01-01'
        }}
      />
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
