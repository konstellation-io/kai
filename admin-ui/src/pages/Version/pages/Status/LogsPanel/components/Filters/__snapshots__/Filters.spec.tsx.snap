// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters matches snapshot 1`] = `
<div
  className="container"
>
  <Left
    className="filterContainer"
  >
    <div
      className="searchFilter"
    >
      <SearchSelect
        className="searchForm"
        hideError={true}
        hideLabel={true}
        onChange={[Function]}
        options={Array []}
        placeholder="Search"
        showSearchIcon={true}
      />
    </div>
    <div
      className="selectProcesses"
    >
      <GroupSelect
        formSelectedOptions={Object {}}
        hideError={true}
        hideSelections={true}
        label=""
        onChange={[Function]}
        options={Object {}}
        placeholder="Select Processes"
      />
    </div>
    <div
      className="selectType"
    >
      <Select
        className="selectTypeForm"
        hideError={true}
        label=""
        onChange={[Function]}
        options={
          Array [
            "DEBUG",
            "ERROR",
            "INFO",
            "WARN",
          ]
        }
        placeholder="ALL LEVELS"
        theme="light"
      />
    </div>
  </Left>
  <Right
    className="filterContainer rightFilters"
  >
    <DateFilter
      selectedOption="2020-01-01"
    />
  </Right>
</div>
`;
